tiledb/api/c_api/attribute/attribute_api_external.h

```diff
@@ -305,7 +305,7 @@ TILEDB_EXPORT int32_t tiledb_attribute_get_cell_size(
  * @param out The output.
  * @return `TILEDB_OK` for success and `TILEDB_ERR` for error./
  */
-TILEDB_EXPORT int32_t tiledb_attribute_dump(
+TILEDB_DEPRECATED_EXPORT int32_t tiledb_attribute_dump(
```

Comment:
As a general observation, while I don't like these APIs because `FILE*` is a `libc`-specific construct, I wouldn't have a problem with keeping them around.

These APIs are easy to maintain and I believe we should one day start having a higher bar for API deprecations and removals, for the sake of ecosystem stability and compatibility.